### ISSUE TITLE
Unable to deploy a record type for an object:  An object 'Contact.<RecordType>' of type RecordType was named in package.xml, but was not found in zipped directory

### ISSUE THREAD

**Issue by matthew-roark** [scolladon/sfdx-git-delta#124]
We are attempting to deploy a new record type for the standard Contact object.  Despite having the new record type in our source for the object, we get the following error message:  An object 'Contact.< RecordType>' of type RecordType was named in package.xml, but was not found in zipped directory

In our build script, we are following the instructions provided for this package.  I have included the relevant snippet below.  The error occurs when we execute the second line.  From our investigation, the sgd package is apparently identifying that the record type is changed and is included in the package.xml, but the package is not correctly gathering the record type to pull into the zipped directory created as part of the  sgd:source:delta command.

// Get the delta changes for the branch
sfdx sgd:source:delta --to HEAD --from HEAD^ --output .
// Deploy the delta changes
sfdx force:source:deploy -u <USERNAME_OMITTED> -x package/package.xml

**Comment by scolladon** [comment]
Hi @matthew-roark!

Thanks for your interest on the plugin and thanks for submitting this issue and by doing so contribute to make this product better!

Though, by reading what you are describing, it seems to me that the delta generation is working fine...
If I understand properly it is the deploy option which is not working as expected.
It could have been the case with any package.xml containing RecordType metadata element I guess, right ?

If yes then maybe it is an issue with the `sfdx force:source:deploy` command itself.
Or maybe RecordType cannot appear in package.xml without another related metadata entity (CustomObject perhaps) and if do you know if there is a documentation somewhere about that behavior ?

**Comment by matthew-roark** [comment]
Hi @scolladon,

We are not seeing this with any package including the RecordType metadata element. In fact, we created another package shortly after this which also had a new RecordType on a different object, and did not see the issue.  

As I stated in my initial entry, we did some basic troubleshooting on our end, and did identify that the changed RecordType was in the repository, and was identified and added to the package.xml by the SGD tool.  What we assumed was that the zipped directory mentioned in the error message was generated in the 'package' directory by the SGD tool along with the package.xml, and this zipped directory is what the deploy command is using to deploy to the server.  In this case, it appears that the file is not being added to this zip package.  

Does this make sense? Or are we assuming a behavior which doesn't exist?

**Comment by scolladon** [comment]
I think I see your point.

The example provided is not using the `-d` parameter of the command then I guess just the delta package is generated, no changes in the existing files, and no delta file deployed, only `-x` from actual sources and delta package generation

Then I have 2 questions :

1. Does the repository is deployable in full ? (using `sfdx force:source:deploy -p <SFDX_SOURCE_FOLDER>` per example)
2. Could you provide reproduction steps to isolate the issue ? You can use the [reproduction playground](https://github.com/scolladon/sfdx-git-delta-reproduction-playground) to create a PR with the reproduction scenario (repository state and command steps to reproduce)

**Comment by matthew-roark** [comment]
@scolladon 

We were able to deploy the repository in full, but were not able to create repro steps.  We did resolve this issue by instead using the -d parameter to get the delta files and deploy this directory instead of using the package.xml.  In case anyone else runs into this issue, I've included our workaround below.

// Create the changed sources directory, and save the changed source files to this directory
mkdir changed-sources && sfdx sgd:source:delta --to HEAD --from HEAD^ --output changed-sources/ -d 
// check if the delta found and created the appropriate source directory for the new/changed files.  if so, deploy this source.  if not, echo to the terminal
$ [ "$(ls -A /changed-sources/force-app/main)" ] && sfdx force:source:deploy -l RunLocalTests -u $(salesforceProdUsername) -p changed-sources/force-app/main || echo "/changed-sources/force-app/main is empty or missing; skipping deploy against SFDC"  
// check if the delta found and created the appropriate source directory for the destructiveChanges.  if so, deploy these changes.  if not, echo to the terminal
$ [ "$(ls -A /changed-sources/destructiveChanges)" ] && sfdx force:mdapi:deploy -u $(salesforceProdUsername) -d destructiveChanges --ignorewarnings || echo "destructiveChanges is empty or missing; skipping deployment to SFDC"
